### ISSUE TITLE
Better error message for 'catch' block missing 'try'.

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#216]
Consider the following source file:

``` TypeScript
catch (x) {
}
```

Currently the compiler reports these errors:

```
(1,2): Declaration or statement expected.
(1,12): ';' expected.                    
```

And if we merge 57d7cf54c696a67017791db09976df4769e1d834, the compiler will report

```
(1,1): Declaration or statement expected.
(1,11): '=>' expected.                   
```

We should give better errors for `catch` blocks missing `try` blocks.
